contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinSynthetix.sol

```diff
@@ -0,0 +1,119 @@
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+interface ISynthetix {
+    // Ethereum Mainnet
+    function exchangeAtomically(
+        bytes32 sourceCurrencyKey,
+        uint256 sourceAmount,
+        bytes32 destinationCurrencyKey,
+        bytes32 trackingCode,
+        uint256 minAmount
+    ) external returns (uint256 amountReceived);
+
+    // Optimism
+    function exchangeWithTracking(
+        bytes32 sourceCurrencyKey,
+        uint256 sourceAmount,
+        bytes32 destinationCurrencyKey,
+        address rewardAddress,
+        bytes32 trackingCode
+    ) external returns (uint256 amountReceived);
+}
+
+interface IReadProxyAddressResolver {
+    function target() external view returns (address);
+}
+
+interface IAddressResolver {
+    function getAddress(bytes32 name) external view returns (address);
+}
+
+contract MixinSynthetix {
+    address private constant rewardAddress =
+        0x5C80239D97E1eB216b5c3D8fBa5DE5Be5d38e4C9;
+    bytes32 constant trackingCode =
+        0x3058000000000000000000000000000000000000000000000000000000000000;
+
+    function _tradeSynthetix(uint256 sellAmount, bytes memory bridgeData)
+        public
+        returns (uint256 boughtAmount)
+    {
+        (
+            IReadProxyAddressResolver readProxy,
+            bytes32 sourceCurrencyKey,
+            bytes32 destinationCurrencyKey
+        ) = abi.decode(
+                bridgeData,
+                (IReadProxyAddressResolver, bytes32, bytes32)
+            );
+
+        boughtAmount = exchange(
+            readProxy,
+            sourceCurrencyKey,
+            destinationCurrencyKey,
+            sellAmount
+        );
+    }
+
+    function exchange(
+        IReadProxyAddressResolver readProxy,
+        bytes32 sourceCurrencyKey,
+        bytes32 destinationCurrencyKey,
+        uint256 sellAmount
+    ) internal returns (uint256 boughtAmount) {
+        ISynthetix synthetix = getSynthetix(readProxy);
+        uint256 chainId;
+        assembly {
+            chainId := chainid()
+        }
+
+        if (chainId == 1) {
+            boughtAmount = synthetix.exchangeAtomically(
+                sourceCurrencyKey,
+                sellAmount,
+                destinationCurrencyKey,
+                trackingCode,
+                0
+            );
+        } else {
+            boughtAmount = synthetix.exchangeWithTracking(
+                sourceCurrencyKey,
+                sellAmount,
+                destinationCurrencyKey,
+                rewardAddress,
+                trackingCode
+            );
+        }
+    }
+
+    function getSynthetix(IReadProxyAddressResolver readProxy)
+        private
+        view
+        returns (ISynthetix)
+    {
+        return
+            ISynthetix(
+                IAddressResolver(readProxy.target()).getAddress("Synthetix")
```

Comment:
It seems unlikely this would change often? And if it is provided than only the quotes during this change would revert if using an old version?

Is it worth just passing in "Synthetix" address and resolving it in the Sampling? Since we're only using the Read proxy to read this address and not for anything else?

Reply:
good point! 

I will update the PR

Reply:
Done!